_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+                break;
+            // END REFACTOR OPTIONS //
+
+            default:
+                // if no special processing, check to see if it is a key in metadata, or failing that, displayDefaults
+                value = (this.#getParameterCaseInsensitive(metadata, token.token) ?? this.#getParameterCaseInsensitive(displayDefaults, token.token)) ?? ""
+                // default formatter is casing
+                formatter = "casing"
+        }
+
+        if (value && (!Array.isArray(value) || value.length > 0)) {
+
+            let finalStr = token.prefix
+            if (formatter == "name") {
+                finalStr += this.#getFormattedName(value, token, metadata)
+            } else if (formatter == "date") {
+                finalStr += this.#getFormattedDate(value, token, targetDate, metadata)
+            } else if (formatter === "casing") {
+                finalStr += this.#getFormattedCaseString(value, token)
+            } else if (formatter == "affiliation-list") {
+                finalStr += this.#getFormattedAffiliationList(value, token)
+            } else if (formatter == "whereabout-list") {
+                finalStr += this.#getFormattedWhereaboutList(value, token, targetDate)
+            }
+            else {
+                finalStr += value
+            }
+
+            finalStr += token.suffix
+
+            return finalStr.trim();
+        } else {
+            return ""
+        }
+    }
+
+    formatDisplayString(input, file, targetDate, overrides) {
+
+        function cleanUpResultString(resultString) {
+            // Normalize spaces (collapse multiple spaces to one) and trim the string
+            resultString = resultString.replace(/\s+/g, ' ').trim();
+        
+            // Remove occurrences of `*( )*` and `( )` (with any number of spaces inside the parentheses)
+            resultString = resultString.replace(/\*\(\s*\)\*|\(\s*\)/g, '');
+        
+            // Clean up spaces around parentheses
+            resultString = resultString.replace(/\(\s+/g, '(').replace(/\s+\)/g, ')');
+        
+            // Trim leading and trailing commas and colons
+            resultString = resultString.replace(/^[:\s,]+|[:\s,]+$/g, '');
+        
+            return resultString;
+        }
+
+        if (this.debug) console.log("Formatting display string: " + input);
+
+        let formattedString = '';
+        let lastIndex = 0;
+    
+        for (let tokenMatch of input.matchAll(this.tokenRegex)) {
+            let tokenStartIndex = input.indexOf(tokenMatch[0], lastIndex);
+    
+            // Append the part of the string before the current token
+            formattedString += input.substring(lastIndex, tokenStartIndex);
+            if (this.debug) console.log("Token match: " + tokenMatch[0] + " at index " + tokenStartIndex + " which is " + (tokenStartIndex === 0));
+    
+            let token = this.#parseTokenString(tokenMatch[0], (tokenStartIndex === 0));
```

Comment:
Is tokenStartIndex == 0 really going to be true when you have a string like `<ancestry> <species>` with no ancestry? Is first needs to be true when either (a) this is the first token or (b) all prior tokens had no values. 

You probably can just check the length of the result string.

Reply:
fixed